### PR TITLE
uses simplified resource interface for lifecycle resource

### PR DESCRIPTION
**Pull request by xh3b4sd**
Nothing big. Just moving code. Logic stays the same. Diff is a bit unfortunate though.

### CODE REVIEW

File: service/awsconfig/v10/resource/lifecycle/create.go [hunk 1]
```diff
@@ -2,12 +2,315 @@ package lifecycle
...
+	"k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	cloudformationservice "github.com/giantswarm/aws-operator/service/awsconfig/v10/cloudformation"
+	"github.com/giantswarm/aws-operator/service/awsconfig/v10/key"
 )
 
-func (r *Resource) ApplyCreateChange(ctx context.Context, obj, createChange interface{}) error {
+func (r *Resource) EnsureCreated(ctx context.Context, obj interface{}) error {
```
Comment (rossf7) [inline service/awsconfig/v10/resource/lifecycle/create.go]: I think GoDoc with a short summary of what's going on would be useful.

File: service/awsconfig/v10/resource/lifecycle/delete.go [hunk 2]
```diff
@@ -2,18 +2,8 @@ package lifecycle
 
 import (
 	"context"
-
-	"github.com/giantswarm/operatorkit/framework"
 )
 
-func (r *Resource) ApplyDeleteChange(ctx context.Context, obj, deleteChange interface{}) error {
+func (r *Resource) EnsureDeleted(ctx context.Context, obj interface{}) error {
```
Comment (rossf7) [inline service/awsconfig/v10/resource/lifecycle/delete.go]: Same here. GoDoc saying delete is a no-op is useful if you're not familiar with the resource.

File: service/awsconfig/v10/resource/lifecycle/create.go (hunk 1 above)
Comment (xh3b4sd) [reply service/awsconfig/v10/resource/lifecycle/create.go]: Done.

File: service/awsconfig/v10/resource/lifecycle/delete.go (hunk 2 above)
Comment (xh3b4sd) [reply service/awsconfig/v10/resource/lifecycle/delete.go]: Done.

**Pull request merged**